test/display_test.py

```diff
@@ -416,20 +449,42 @@ def todo_test_set_palette(self):
 
         self.fail()
 
-    def todo_test_toggle_fullscreen(self):
+    def test_toggle_fullscreen(self):
```

Comment:
toggle_fullscreen() doesn't work with the dummy video driver. We should skip this whole test on CI.

According to the docs it also only works on X11 (linux) - this is clearly untrue looking at the current code which indicates support varies widely on a per platform basis. Probably we should also raise an issue to try and update the docs with more accurate info on what platforms toggle_fullscreen() now works on.

Two approaches here, you could either parse through the C code to try and work out which platforms currently support toggle_fullscreen(). This is the relevant part of the code:

https://github.com/pygame/pygame/blob/7d3900d5b04dc8102f218cc60f3fbf5e06dd8fa1/src_c/display.c#L2515-L2559

And then add checking for those platforms as part of the test (looks like it should work on windows, mac (cocoa) and linux (x11 & wayland drivers) at least but fail on android which is always full screen). Or, probably better, you could just make the test fail if the function raises an unexpected exception (as I mention below) and then build up a skip list of list of platforms that the test would fail on.

To figure out platforms for this you can probably mostly rely on the SDL_VIDEODRIVER environment variable again which also has 'android' in addition to the 'dummy' on used on CI. See this list in the get_driver test:

https://github.com/pygame/pygame/blob/48e94ef6106f3c1b79f95e0012f8f7ac2a964066/test/display_test.py#L106-L137

Reply:
TLDR: skip the test on dummy SDL_VIDEODRIVER and android at least.